src/app/components/category-products/category-products.component.ts

```diff
@@ -43,7 +43,10 @@ export class CategoryProductsComponent {
   ])
 
   readonly sortingOption: FormControl = new FormControl();
-  readonly sortingOption$: Observable<string | null> = this.sortingOption.valueChanges.pipe(startWith(null));
+  readonly sortingOption$: Observable<string | null> = this.sortingOption.valueChanges.pipe(startWith(null), shareReplay(1));
```

Comment:
It's not good practice to give the same name and only add "$" mark. It should be something like: sortingOptionValues or something like that